src/backend/api/controllers/camera.js

```diff
@@ -0,0 +1,81 @@
+const WebSocket = require('ws');
+const config = require('config');
+const rclnodejs = require('rclnodejs');
+
+class CameraController {
+    constructor() {
+        this.startCameraWS = this.startCameraWS.bind(this);
+        this.onConnection = this.onConnection.bind(this);
+        // this.onError = this.onError.bind(this);
+        // this.onClose = this.onClose.bind(this);
+        this.socket = null;
+        this.camera_node = null;
+    }
+
+    async startCameraController() {
+        await rclnodejs.init();
+        this.camera_node = new rclnodejs.Node('camera_node');
+        this.cameraSubscriber = this.camera_node.createSubscription(
+            'sensor_msgs/msg/CompressedImage',
+            '/camera_feed',
+            (msg) => this.video_frames_callback(msg)
+        );
+        console.log('Camera controller started');
+        this.camera_node.spin();
+    }
+
+    async onConnection(ws) {
+        ws.on('message', data => this.onMessage(ws, data));
+        ws.on('error', error => this.onError(ws, error));
+        ws.on('close', () => this.onClose(ws));
+        console.log('WebSocket connection established');
+        this.startCameraController();
+    }
+
+
+    async video_frames_callback(msg) {
+        console.log('Received video frame');
+        const imageData = Buffer.from(msg.data); // Convert data to Buffer
+        const base64Image = imageData.toString('base64'); // Encode to Base64
+        if (this.socket) {
+          this.socket.clients.forEach((ws) => ws.send(base64Image));
+        }
+      }
```

Comment:
aqui acho que da pra verificar o `if (this.socket)` lá no começo e já meter um early return, pq se n tiver socket nem precisa encodar nada